### PR TITLE
Fixed a JS console warning, react-router and react-bootstrap

### PR DESCRIPTION
**Pull request by aliceriot**
#### What are the relevant tickets?

none, this fixes a bug I noticed.

#### What's this PR do?

this adds a library called react-router-bootstrap that is designed to
make react-router and react-bootstrap play nicely together.

basically, we were getting a JS warning that looks like this:

```
warning.js:45 Warning: validateDOMNesting(...): <a> cannot appear as a descendant of <a>. See LoginButton > MenuItem > SafeAnchor > a > ... > Link > a
```

this is occurring because we had the following code:

```javascript
<MenuItem>
  <Link to="/profile">
    Profile
  </Link>
</MenuItem>
```

`MenuItem` comes from `react-bootstrap`, and `Link` is a `react-router`
link. basically, the problem is that `MenuItem` implicitly includes an
`<a ..></a>` element, and places `this.props.children` inside that
element.

hence, we get a console warning about improperly nesting `<a>` tags.

of course, the outer one has `href=undefined`, so only the inner
`<Link>` tag does anything, and the ui works as expected, but still, the
warning.

the `react-router-bootstrap` library saves us here. we get a component
called `<LinkContainer>` which can wrap any `react-bootstrap` component
and make it behave like a `react-router` `<Link>`. we just set a `to`
prop on the `<LinkContainer>` with the path we want and any url params,
and off we go!

#### Where should the reviewer start?

I think just take a look in here:

	modified:   static/js/containers/LoginButton.js

#### How should this be manually tested?

pull down the branch, restart docker, visit `/dashboard` and confirm the
js console error is gone.

#### Any background context you want to provide?

docs for the lib here:
https://github.com/react-bootstrap/react-router-bootstrap

### CODE REVIEW

File: static/js/containers/LoginButton.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import { connect } from 'react-redux';
 import DropdownButton from 'react-bootstrap/lib/DropdownButton';
+import { LinkContainer } from 'react-router-bootstrap';
```
Comment (noisecapella) [inline static/js/containers/LoginButton.js]: Can you use this syntax instead?

`import LinkContainer from 'react-router-bootstrap/lib/LinkContainer';`

Importing modules directly is best in case there's other stuff in `react-router-bootstrap` which webpack can leave out

Comment (aliceriot) [inline static/js/containers/LoginButton.js]: sure

**Comment by noisecapella** [comment]
Great find, I was wondering where this came from!

**Comment by aliceriot** [comment]
@noisecapella I ran into this on a previous project, so I knew where to look!

File: static/js/containers/LoginButton.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
...
 import MenuItem from 'react-bootstrap/lib/MenuItem';
 import Link from 'react-router/lib/Link';
```
Comment (noisecapella) [inline static/js/containers/LoginButton.js]: Can you remove this line since we don't use `Link` anymore?

Comment (aliceriot) [inline static/js/containers/LoginButton.js]: what do you think about adding the `no-unused-vars` rule to `eslint.rc`? I think that should make noise about unused imports.

Comment (noisecapella) [inline static/js/containers/LoginButton.js]: Sure, I kind of assumed this was a default rule but apparently not. Can we do this in a separate PR?

Comment (aliceriot) [inline static/js/containers/LoginButton.js]: oof nvm it's waaaay loud, it complains about this:

```
`import React from 'react';`

class Foo extends React.Component {}
```

says react isn't used...

Comment (noisecapella) [inline static/js/containers/LoginButton.js]: `React` is a strange one because every `<div />` gets translated to a `React.createElement("div")` which requires `React`, but you don't see it in the original code. It looks like we can filter out certain errors based on variable name: http://eslint.org/docs/rules/no-unused-vars#caughterrorsignorepattern

Comment (aliceriot) [inline static/js/containers/LoginButton.js]: ooh nifty!

I'll open another pr and fiddle with this!

**Comment by noisecapella** [comment]
Functionality looks good but I see this blue background now when I'm on the user profile:

screenshot from 2016-04-15 13-48-10

We can deal with it in another PR if you prefer :+1:

**Comment by aliceriot** [comment]
@noisecapella should be fixed! just a quick prop change.

**Comment by noisecapella** [comment]
Looks good!
